overcloud.yml

```diff
@@ -257,14 +257,26 @@
 
       - name: set facts for ceph deployment
         set_fact:
-          ceph_params: "--storage-backend ceph --storage-external no"
+          ceph_params: "{{ (osp_release|int >= 17) | ternary('--storage-backend ceph --storage-external no --ceph-osd-spec-file {{ osd_spec_file }}','--storage-backend ceph --storage-external no') }}"
         when: ceph_enabled
 
       - name: set deployment_timeout to 2400
         set_fact:
           deployment_timeout: 2400
         when: scale_compute_vms == true
 
+      - block:
+        - name: set fact for osd-spec.yml file
+          set_fact:
+            osd_spec_file: "{{ ansible_user_dir }}/osd-spec.yml"
```

Comment:
shouldn't  osd_spec_file fact set before `set facts for ceph deployment` task as it will fail otherwise
https://github.com/redhat-performance/jetpack/pull/497/files#diff-962dc7050a566bd835f3065a4da65d205c5ca498df81ec4262079970ee417fd7R258-R260

Reply:
ignore my comment before as the value is string ceph_params

Reply:
okay, i can still move it up if you suggest 👍